src/extension-runners/chromium.js

```diff
@@ -151,12 +151,37 @@ export class ChromiumExtensionRunner {
     });
   }
 
-  wssBroadcast(data: Object) {
-    for (const client of this.wss.clients) {
-      if (client.readyState === WebSocket.OPEN) {
-        client.send(JSON.stringify(data));
+  async wssBroadcast(data: Object) {
+    return new Promise((resolve) => {
+      const clients = this.wss ? new Set(this.wss.clients) : new Set();
+
+      if (clients.size === 0) {
+        resolve();
       }
-    }
+
+      const webExtReloadComplete = async (message) => {
+        const msg = JSON.parse(message);
+
+        if (msg.type === 'webExtReloadExtensionComplete') {
+          for (const client of clients) {
+            client.removeListener('message', webExtReloadComplete);
+          }
+          resolve();
+        }
+      };
+
+      for (const client of clients) {
+        if (client.readyState === WebSocket.OPEN) {
+          client.on('message', webExtReloadComplete);
+          client.on('close', () => {
+            client.off('message', webExtReloadComplete);
+            clients.delete(client);
+          });
+
+          client.send(JSON.stringify(data));
+        }
```

Comment:
Add an `else` block that calls `clients.delete(client);`, and move the `if (clients.size === 0) {` check from above to after the loop.